apps/fz_http/test/fz_http_web/controllers/json/user_controller_test.exs

```diff
@@ -50,41 +65,207 @@ defmodule FzHttpWeb.JSON.UserControllerTest do
+  describe "[authed] GET /v0/users/:id" do
+    setup _tags, do: {:ok, conn: authed_conn()}
+
+    test "gets user by id", %{conn: conn} do
+      user = conn.private.guardian_default_resource
+      conn = get(conn, ~p"/v0/users/#{user}")
+      assert json_response(conn, 200)["data"]["id"] == user.id
+    end
+
+    test "gets user by email", %{conn: conn} do
+      user = conn.private.guardian_default_resource
+      conn = get(conn, ~p"/v0/users/#{user.email}")
+      assert json_response(conn, 200)["data"]["id"] == user.id
+    end
+
+    test "renders 404 for user not found", %{conn: conn} do
+      assert_error_sent 404, fn ->
+        get(conn, ~p"/v0/users/003da73d-2dd9-4492-8136-3282843545e8")
+      end
+    end
+  end
+
+  describe "[authed] DELETE /v0/users/:id" do
+    setup _tags, do: {:ok, conn: authed_conn()}
+
+    test "deletes user by id", %{conn: conn} do
+      user = UsersFixtures.user(%{role: :unprivileged})
       conn = delete(conn, ~p"/v0/users/#{user}")
       assert response(conn, 204)
 
       assert_error_sent 404, fn ->
         get(conn, ~p"/v0/users/#{user}")
       end
     end
+
+    test "deletes user by email", %{conn: conn} do
+      user = UsersFixtures.user(%{role: :unprivileged})
+      conn = delete(conn, ~p"/v0/users/#{user.email}")
+      assert response(conn, 204)
+
+      assert_error_sent 404, fn ->
+        get(conn, ~p"/v0/users/#{user}")
+      end
+    end
+
+    test "renders 404 for user not found", %{conn: conn} do
+      assert_error_sent 404, fn ->
+        delete(conn, ~p"/v0/users/003da73d-2dd9-4492-8136-3282843545e8")
+      end
+    end
+  end
+
+  describe "[unauthed] GET /v0/users/:id" do
+    setup _tags, do: {:ok, conn: unauthed_conn()}
+
+    test "renders 401 for missing authorization header", %{conn: conn} do
+      conn = get(conn, ~p"/v0/users/invalid")
+      assert json_response(conn, 401)["errors"] == %{"auth" => "unauthenticated"}
+    end
+  end
+
+  describe "[unauthed] PUT /v0/users/:id" do
+    setup _tags, do: {:ok, conn: unauthed_conn()}
+
+    test "renders 401 for missing authorization header", %{conn: conn} do
+      conn = put(conn, ~p"/v0/users/invalid", user: %{})
+      assert json_response(conn, 401)["errors"] == %{"auth" => "unauthenticated"}
+    end
+  end
+
+  describe "[unauthed] GET /v0/users" do
+    setup _tags, do: {:ok, conn: unauthed_conn()}
+
+    test "renders 401 for missing authorization header", %{conn: conn} do
+      conn = get(conn, ~p"/v0/users")
+      assert json_response(conn, 401)["errors"] == %{"auth" => "unauthenticated"}
+    end
+  end
+
+  describe "[unauthed] POST /v0/users" do
+    setup _tags, do: {:ok, conn: unauthed_conn()}
+
+    test "renders 401 for missing authorization header", %{conn: conn} do
+      conn = post(conn, ~p"/v0/users", user: %{})
+      assert json_response(conn, 401)["errors"] == %{"auth" => "unauthenticated"}
+    end
+  end
+
+  describe "[unauthed] DELETE /v0/users/:id" do
```

Comment:
I would generally not split describe blocks to keep things related to each other, otherwise, somebody can insert another test in between them. Those blocks don't need a `setup` because you can use `authed_conn/1`/`unauthed_conn/1` as the first argument of the `post`/`get`/`delete` functions.

Reply:
That's funny, I had started to do that, but thought you might prefer this more. I'll update.

Reply:
Fixed